File: Examples/CSharp/WorkingWithProjects/Miscellaneous/CalculateCriticalPath.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects.Miscellaneous
{
    using System;

    public class CalculateCriticalPath
    {
        public static void Run()
        {
            // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            //ExStart:CalculateCriticalPath
            var project = new Project(dataDir + "CalculateCriticalPath.mpp");
            project.CalculationMode = CalculationMode.Automatic;

            var subtask1 = project.RootTask.Children.Add("1");
            var subtask2 = project.RootTask.Children.Add("2");
            project.TaskLinks.Add(subtask1, subtask2, TaskLinkType.FinishToStart);

            project.RootTask.Children.Add("3");

            // Display the critical path now
            foreach (var task in project.CriticalPath)
            {
                Console.WriteLine(task.Get(Tsk.Name));
            }
            //ExEnd:CalculateCriticalPath
        }
    }
}

File: Examples/VisualBasic/WorkingWithProjects/Miscellaneous/CalculateCriticalPath.vb
Imports System

Namespace WorkingWithProjects.Miscellaneous
    Public Class CalculateCriticalPath
        Public Shared Sub Run()
            ' ExStart: CalculateCriticalPath
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            Dim project As New Project(dataDir & "CalculateCriticalPath.mpp")
            project.CalculationMode = CalculationMode.Automatic

            Dim subtask1 As Task = project.RootTask.Children.Add("1")
            Dim subtask2 As Task = project.RootTask.Children.Add("2")
            Dim subtask3 As Task = project.RootTask.Children.Add("3")
            project.TaskLinks.Add(subtask1, subtask2, TaskLinkType.FinishToStart)

            ' Display the critical path now
            For Each task As Task In project.CriticalPath
                Console.WriteLine(task.Get(Tsk.Name))
            Next task
            ' ExEnd: CalculateCriticalPath
        End Sub
    End Class
End Namespace